### PR TITLE
Generate command with new structure

### PR DESCRIPTION
**Pull request by ferrannp**
- You pass now an `entryPoint` using -E if you want to use predefined generators (component, container and reducer)
- Types
- Tests

### CODE REVIEW

File: packages/gluestick/src/generator/generator.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ * @param command {Object} { generatorName, entityName, entryPoint, options } Command object
+ * @param logger {Object} logger Logger instance
+ */
+module.exports = (command: Command, logger: Logger): void => {
+  const {
+    generatorName,
+    entityName,
+    options = {},
+  } = command;
```
Comment (michalchudziak) [inline packages/gluestick/src/generator/generator.js:34]: Destruct in args definition

File: packages/gluestick/src/generator/generator.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const predefinedGenerators = [
+    'component',
+    'reducer',
+    'container',
+  ];
+
+  if (predefinedGenerators.find((element) => element === generatorName)) {
+    if (!isValidEntryPoint(entryPoint, logger)) {
+      return;
```
Comment (michalchudziak) [inline packages/gluestick/src/generator/generator.js:47]: Can't we throw an error here?

File: packages/gluestick/src/generator/generator.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+
+  if (!/^[a-zA-Z0-9/_-]+$/.test(entityName)) {
+    logger.error('Invalid name specified');
+    return;
```
Comment (michalchudziak) [inline packages/gluestick/src/generator/generator.js:53]: Shouldn't we also kill process here?

**Review by michalchudziak — CHANGES_REQUESTED** [review]

File: packages/gluestick/src/generator/generator.js (hunk 3 above)
Comment (ferrannp) [reply packages/gluestick/src/generator/generator.js:53]: I think we should not kill the process (see https://www.nczonline.net/blog/2014/02/04/maintainable-node-js-javascript-avoid-process-exit/) but instead throw errors. 

In this case though, it is not like a code error but a error message (using their `logger.error`) to the user when typing the command. I think in this case it's better this way although I am not sure 100% sure.

File: packages/gluestick/src/generator/generator.js (hunk 2 above)
Comment (ferrannp) [reply packages/gluestick/src/generator/generator.js:47]: Same comment as the previous `return`. `isValidEntryPoint` already logs errors + info.

File: packages/gluestick/src/generator/generator.js (hunk 1 above)
Comment (ferrannp) [reply packages/gluestick/src/generator/generator.js:34]: This is probably a matter of test, for me it is easier to read the function this way.

**Review by michalchudziak — APPROVED** [review]

**Pull request merged**